fiftyone/core/collections.py

```
@@ -10548,42 +10523,115 @@ def create_index(self, field_or_spec, unique=False, wait=True, **kwargs):
+            # unique like other manually created indexes, but they are, so
+            # nothing needs to be done here
             return index_name
 
-        # Setting `w=0` sets `acknowledged=False` in pymongo
-        write_concern = None if wait else WriteConcern(w=0)
+        # Handle default indexes
+        if (
+            _index_name in self._get_default_indexes(frames=is_frame_index)
+            and index_name != "filepath"  # allow 'filepath' to be modified
+        ):
+            raise ValueError(f"Cannot modify default index '{index_name}'")
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)